### PR TITLE
Support for modifying incoming value when writing on ESIL register

### PR DESCRIPTION
**Pull request by killabytenow**
Modification on operation `hook_reg_write()` in `RAnalEsilCallbacks` struct, for supporting modification of the incoming value,

### CODE REVIEW

**Comment by radare** [comment]
dude, dont merge in PRs

File: libr/anal/p/anal_8051.c [hunk 1]
```diff
@@ -401,14 +401,14 @@ static int i8051_hook_reg_read(RAnalEsil *esil, const char *name, ut64 *res, int
...
-static int i8051_hook_reg_write(RAnalEsil *esil, const char *name, ut64 val) {
+static int i8051_hook_reg_write(RAnalEsil *esil, const char *name, ut64 *val) {
 	int ret = 0;
 	struct r_i8015_reg *ri;
 	RAnalEsilCallbacks cbs = esil->cb;
 
 	if ((ri = i8051_reg_find (name))) {
 		ut8 offset = i8051_reg_get_offset(esil, ri);
-		ret = r_anal_esil_mem_write (esil, IRAM + offset, (ut8*)&val, ri->num_bytes);
+		ret = r_anal_esil_mem_write (esil, IRAM + offset, (ut8*) val, ri->num_bytes);
```
Comment (radare) [inline libr/anal/p/anal_8051.c]: unnecessary space

**Comment by radare** [comment]
im afraid of this api change for two reasons:

* the only way the user have to know that the value that has passed has been modified by the callee is by storing a copy of the old value and checking it again after the call.
* Who modifies this value? a hook? this seems inconsistent to me because the order of the calls may result in undesired behaviours

**Comment by killabytenow** [comment]
I think that the idea of a hook is to monitorize and take control of the situation for changing the standard execution flow, so it is normal that in case of using several hooks not only the results may be impredictable, but the order always will be critical.

But in this case it is not possible to set two hooks, at least without being tricky, because this entry is not a list of hooks, but a single callback pointer, so by default we cannot think about calllbacks' order processing.

IMHO this solution is not too bad, but other solution is introducing new callbacks (i.e. preset, postset) to avoid updating external bindings or third party code (if it exists).

File: libr/anal/p/anal_8051.c (hunk 1 above)
Comment (killabytenow) [inline libr/anal/p/anal_8051.c]: Fixed in new commit.

**Comment by radare** [comment]
please clean your branch. drop those merges and squash that

**Comment by killabytenow** [comment]
Superseded by PR https://github.com/radare/radare2/pull/5977.

**Pull request closed without merging**